cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -1098,6 +1123,21 @@ Status TransferDecimal(RecordReader* reader, MemoryPool* pool,
   return Status::OK();
 }
 
+Status TransferExtension(RecordReader* reader, std::shared_ptr<DataType> value_type,
+                         const ColumnDescriptor* descr, MemoryPool* pool, Datum* out) {
+  std::shared_ptr<ChunkedArray> result;
+  auto ext_type = std::static_pointer_cast<::arrow::ExtensionType>(value_type);
+  auto storage_type = ext_type->storage_type();
+  RETURN_NOT_OK(TransferColumnData(reader, storage_type, descr, pool, &result));
+
+  auto result0 = result->chunk(0);
+  auto ext_data = result0->data()->Copy();
+  ext_data->type = ext_type;
+  auto ext_result = ext_type->MakeArray(ext_data);
```

Comment:
Is there a generic way to create a create a ChunkedArray of a certain type? Or will I need to loop through the chunks, create an ExtensionArray for each chunk, and create a ChunkedArray of all converted chunks?

(I suppose we should implement `View()` for viewing an array as an extensionarray)

Reply:
The latter.

Reply:
I replaced the above with

```
  ::arrow::ArrayVector out_chunks(result->num_chunks());
  for (int i = 0; i < result->num_chunks(); i++) {
     auto chunk = result->chunk(i);
     auto ext_data = chunk->data()->Copy();
     ext_data->type = ext_type;
     auto ext_result = ext_type->MakeArray(ext_data);
     out_chunks.push_back(ext_result);
  }
  *out = std::make_shared<ChunkedArray>(out_chunks, ext_type);
```

which builds but when running segfaults on the `std::make_shared<ChunkedArray>(out_chunks, ext_type)` line, and can't directly figure out why. Do you see anything obviously wrong ?

Reply:
That's because `out_chunks(result->num_chunks())` already creates a N-entry vector, then with `push_back` you append _after_ the N first entries (which are null by default).

Just replace `out_chunks.push_back(ext_result);` with `out_chunks[i] = ext_result;`